config/config.go

```diff
@@ -109,26 +112,27 @@ type AnalyzersList struct {
 	AggregateStats   *AggregateStatsConfig   `koanf:"aggregate_stats"`
 }
 
-// NodeConfig is the configuration for chain analyzers.
-type NodeConfig struct {
-	// ChainID is the chain ID of the chain this analyzer will process.
-	// Used to identify the chain in the database.
-	ChainID string `koanf:"chain_id"`
-
-	// RPC is the node endpoint.
-	RPC string `koanf:"rpc"`
+// SourceConfig has some controls about what chain we're analyzing and how to connect.
+type SourceConfig struct {
+	// ChainName is the name of the chain (e.g. mainnet/testnet/local).
+	ChainName string `koanf:"chain_name"`
 
-	// ChainContext is the domain separation context.
-	// It uniquely identifies the chain; it is derived as a hash of the genesis data.
-	// Used as safety check to prevent accidental use of the wrong RPC endpoint.
-	ChainContext string `koanf:"chaincontext"`
+	// Nodes describe the oasis-node(s) to connect to. Keys are "archive
+	// names," e.g. "damask."
+	Nodes map[string]*NodeConfig `koanf:"nodes"`
 
 	// If set, the analyzer will skip some initial checks, e.g. that
-	// `rpc` really serves the chain with `chain_context`.
+	// `rpc` really serves the chain with the chain context we expect.
 	// NOT RECOMMENDED in production; intended for faster testing.
 	FastStartup bool `koanf:"fast_startup"`
```

Comment:
I'm OK to get rid of this BTW, and just default to true (= never check).
I don't think this makes it harder to misconfigure a mainnet analyzer to query a testnet node, or vice versa. If you accidentally use a testnet config (via partial node config copy-paste, or via using the wrong config file) on a mainnet analyzer, you'll use it wholesale, with the chain name _and_ context at the same time.
I expect the overwhelming scenario in which the name and context will disagree will be when you change the name intentionally but forgot to hunt down the corresponding chain context.

Reply:
:+1: but let's consider that outside this PR